flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/failover/FailoverRegion.java

```diff
@@ -73,10 +79,21 @@ public FailoverRegion(
 
 		this.executionGraph = checkNotNull(executionGraph);
 		this.connectedExecutionVertexes = checkNotNull(connectedExecutions);
+		this.tasks = initTasks(connectedExecutionVertexes);
```

Comment:
We could wrap this into an `UnmodifiableMap` because it is supposed to be only manipulated once in the constructor, but the field is exposed through a getter.

Reply:
I have one more suggestion that came to my mind from a second look: we could pass in `Map<JobVertexID, ExecutionJobVertex> tasks` directly into the constructor, doing the work of `initTasks` outside of the class before the constructor is invoked. Constructors that only do assignments are typically easier to test and also separates the concern of object construction from the object itself.